web/src/main/java/com/marklogic/hub/web/service/FlowManagerService.java

```diff
@@ -217,48 +214,17 @@ public StepModel createStep(String flowName, Integer stepOrder, String stepId, S
-                Step defaultIngest = defaultIngestDefinition.transformToStep(step.getName(), defaultIngestDefinition, new Step());
-                step = StepModel.mergeFields(stepModel, defaultIngest, step);
-                break;
-            case MAPPING:
-                StepDefinition defaultMappingDefinition = getDefaultStepDefinitionFromResources("hub-internal-artifacts/step-definitions/mapping/marklogic/default-mapping.step.json", StepDefinition.StepDefinitionType.MAPPING);
-                Step defaultMapping = defaultMappingDefinition.transformToStep(step.getName(), defaultMappingDefinition, new Step());
-                step = StepModel.mergeFields(stepModel, defaultMapping, step);
-                break;
-            case MASTERING:
-                StepDefinition defaultMasterDefinition = getDefaultStepDefinitionFromResources("hub-internal-artifacts/step-definitions/mastering/marklogic/default-mastering.step.json", StepDefinition.StepDefinitionType.MASTERING);
-                Step defaultMaster = defaultMasterDefinition.transformToStep(step.getName(), defaultMasterDefinition, new Step());
-                step = StepModel.mergeFields(stepModel, defaultMaster, step);
-                break;
-            case CUSTOM:
-                if (stepDefinitionManagerService.getStepDefinition(step.getStepDefinitionName(), step.getStepDefinitionType()) != null) {
-                    StepDefinition oldStepDefinition = stepDefinitionManagerService.getStepDefinition(step.getStepDefinitionName(), step.getStepDefinitionType());
-                    StepDefinition stepDefinition = oldStepDefinition.transformFromStep(oldStepDefinition, step);
-                    stepDefinitionManagerService.saveStepDefinition(stepDefinition);
-                }
-                else {
-                    String stepDefName = step.getStepDefinitionName();
-                    StepDefinition.StepDefinitionType stepDefType = StepDefinition.StepDefinitionType.CUSTOM;
-                    String modulePath = "/custom-modules/" + stepDefType.toString().toLowerCase() + "/" + stepDefName + "/main.sjs";
-
-                    StepDefinition stepDefinition = StepDefinition.create(stepDefName, stepDefType);
-                    stepDefinition = stepDefinition.transformFromStep(stepDefinition, step);
-
-                    scaffolding.createCustomModule(stepDefName, stepDefType.toString());
-                    stepDefinition.setModulePath(modulePath);
-                    step.setModulePath(modulePath);
-
-                    stepDefinitionManagerService.createStepDefinition(stepDefinition);
-                }
+        boolean invalidStepType = true;
+        for (StepDefinition.StepDefinitionType stepType : StepDefinition.StepDefinitionType.values()) {
+            if (stepType.equals(step.getStepDefinitionType())) {
+                invalidStepType = false;
```

Comment:
Just a nitpick, we can get rid of `invalidStepType` flag and throw the exception from here if we negate the if condition.